tests/virt_autotest/login_console.pm

```diff
@@ -35,9 +35,12 @@ sub login_to_console() {
     assert_screen([qw(grub2 grub1)], 120);
     if (!get_var("reboot_for_upgrade_step")) {
         if (get_var("XEN") || check_var("HOST_HYPERVISOR", "xen")) {
-            send_key_until_needlematch("virttest-bootmenu-xen-kernel", 'down', 10, 1);
-            send_key 'ret';
+            if (!match_has_tag("virttest-bootmenu-xen-kernel")) {
+                send_key 'down';
```

Comment:
* Logic is correct and needed, reasons are:
  First, you should see it is for xen only(kvm is the default first menuentry, no send key is needed). And then we check it is not the desired one with needle 'virttest-bootmenu-xen-kernel', so we send key down to stop grub timeout.
  Why not use original send_key_until_needle match is that most times the grub times out before the ipmi machine responds to the send_key down event, so the default first menuentry is booted. This results a lot of failures in our test. This is also why I call it a code enhancement. The original code logic is correct, but it just does not work in the real test over ipmi.

* But I will change code to use send_key 'up' as grub_test.pm do, since it is more simple, and do not change anything in effect except stop grub timeout